packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/DiagramFilterEventProcessorFactory.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.forms;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.collaborative.api.IRepresentationConfiguration;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationEventProcessor;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationEventProcessorFactory;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationRefreshPolicyRegistry;
+import org.eclipse.sirius.components.collaborative.api.ISubscriptionManagerFactory;
+import org.eclipse.sirius.components.collaborative.api.RepresentationEventProcessorFactoryConfiguration;
+import org.eclipse.sirius.components.collaborative.forms.api.DiagramFilterConfiguration;
+import org.eclipse.sirius.components.collaborative.forms.api.FormCreationParameters;
+import org.eclipse.sirius.components.collaborative.forms.api.IDiagramFilterDescriptionProvider;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormEventHandler;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormEventProcessor;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormPostProcessor;
+import org.eclipse.sirius.components.collaborative.forms.api.IWidgetSubscriptionManagerFactory;
+import org.eclipse.sirius.components.collaborative.forms.configuration.FormEventProcessorConfiguration;
+import org.eclipse.sirius.components.collaborative.forms.configuration.FormEventProcessorFactoryConfiguration;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.forms.description.FormDescription;
+import org.eclipse.sirius.components.forms.renderer.IWidgetDescriptor;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to create the diagram filter event processors.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DiagramFilterEventProcessorFactory implements IRepresentationEventProcessorFactory {
+
+    private final Optional<IDiagramFilterDescriptionProvider> diagramFilterDescriptionProvider;
```

Comment:
And you will thus be able to remove the optional part of this field and constructor argument

Reply:
In the near future, the code for `PropertiesEventProcessorFactory`, `RelatedElementsEventProcessorFactory` and `RepresentationsEventProcessorFactory` and their configurations, GraphQL definitions and frontend components. will move too in Sirius Web. Those are Sirius Web specific features.

Reply:
I moved the classes in `sirius-web-application`. I put them in the same package as the other classes related to this feature: `o.e.s.w.a.diagram.services.filter`. Let me know if this isn't the right place.

The optional wasn't needed anymore, it was there because I initially provided an implementation of `IDiagramFilterDescriptionProvider` for the new architecture, but not the old one. I should have removed it when I added the implementation for the old one.

This being said, I agree that it's odd to have something named "DiagramXXX" in forms. I didn't see the issue of consumer needing to provide something for a feature they don't want, because to me it looked similar for `PropertiesEventProcessorFactory` or `RelatedElementsEventProcessorFactory`, for example. Since all of them will be moved to Sirius Web I see the issue, thanks for the additional info 👍